tests/unit/scan.tcl

```diff
@@ -430,6 +430,29 @@ proc test_scan {type} {
             }
         }
     }
+
+    test "{$type} SCAN MATCH pattern implies cluster slot" {
+        # Tests the code path for an optimization for patterns like "{foo}-*"
+        # which implies that all matching keys belong to one slot.
+        r flushdb
+        for {set j 0} {$j < 100} {incr j} {
+            r set "{foo}-$j" "foo"
+            r set "{bar}-$j" "bar"
```

Comment:
Could we set some data on a slot which is after {foo}/`12182`? Given our cursor moves forward after it's done with a slot.

Reply:
Sure, I can do that, but even if the cursor moves forward, the result is filtered by the pattern afterwards anyway, so this test wouldn't catch it anyway. This test is just for coverage of the new code.